agent/xds/endpoints.go

```diff
@@ -324,7 +325,6 @@ func (s *Server) endpointsFromDiscoveryChain(
 			)
 			if actualTargetID != targetID {
 				targetID = actualTargetID
-				target = chain.Targets[actualTargetID]
```

Comment:
The use of `target` was removed in 8e22d80e3550592a32144f2d1c84ebf6becb957b.

I'm wondering if there is another bug here. `target.Name` is used to create `clusterName` above on 313, but when we change the targetID we never change the clusterName. Is that correct?

Reply:
I think that is still correct. The cluster name should be the name of the "upstream" regardless of redirects and failures which is what the actual target id is figuring out.